spec/requests/admin/lead_providers_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Admin::LeadProviders", type: :request do
+  let(:lead_provider) { FactoryBot.create(:lead_provider) }
+  let(:lead_provider_name) { Faker::Company.name }
+
+  describe "GET /admin/lead_providers" do
+    it "renders the correct template" do
+      # When
+      get "/admin/lead_providers"
+
+      # Then
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "POST /admin/lead_providers" do
```

Comment:
Having every check in a separate request test though tidy is probably not a great idea. It's Ok for now as our test suite is small but as it gets bigger it will become painfully slow for the test suite to complete. For the easy checks you are probably better off combining them in future.

Reply:
I don't know what the overhead is, but each test is running in about 60ms at the moment, so I figured there was no harm

Reply:
As I said, it's Ok for now as the project is very young  but let's keep that in mind in future. Things will probably be different in 3 to 5 years.  I've had to refactor a few test suites due to tests taking more than 45 minutes to run.